src/engine/routing_algorithms/many_to_many.cpp

```diff
@@ -282,15 +324,15 @@ void forwardRoutingStep(const DataFacade<Algorithm> &facade,
                 current_duration = std::min(current_duration, new_duration);
             }
         }
-        else if (new_weight < current_weight)
+        else if (std::tie(new_weight, new_duration) < std::tie(current_weight, current_duration))
         {
             current_weight = new_weight;
             current_duration = new_duration;
         }
     }
 
     relaxOutgoingEdges<FORWARD_DIRECTION>(
-        facade, node, source_weight, source_duration, query_heap, phantom_node);
+        facade, query_heap, node, source_weight, source_duration, phantom_node);
```

Comment:
any specific reason why you changed the order here? -- fine with the change, but curious

Reply:
no, i will revert this particular change